pallets/ah-migrator/src/account.rs

```diff
@@ -169,40 +169,61 @@ impl<T: Config> Pallet<T> {
 	pub fn finish_accounts_migration(rc_balance_kept: T::Balance) -> Result<(), Error<T>> {
 		use frame_support::traits::Currency;
 		let checking_account = T::CheckingAccount::get();
-		// current value is the migrated checking balance of RC
-		let rc_balance_before =
+		let balances_before = AhBalancesBefore::<T>::get();
+		// current value is the AH checking balance + migrated checking balance of RC
+		let checking_balance =
 			<<T as pallet::Config>::Currency as Currency<_>>::total_balance(&checking_account);
-		let ah_issuance_before = AhTotalIssuanceBefore::<T>::get();
 
+		/* Arithmetics explanation:
+		At this point, because checking account was completely migrated:
+			`checking_balance` = ah_check_before + rc_check_before
+			(0) rc_check_before = `checking_balance` - ah_check_before
+
+		Invariants:
+			(1) rc_check_before = sum_total_before(ah, bh, collectives, coretime, people)
+			(2) rc_check_before = sum_total_before(bh, collectives, coretime, people) + ah_total_before
+		Because teleports are disabled for RC and AH during migration, we can say:
+			(3) sum_total_before(bh, collectives, coretime, people) = sum_total_after(bh, collectives, coretime, people)
+		Ergo use (3) in (2):
+			(4) rc_check_before = sum_total_after(bh, collectives, coretime, people) + ah_total_before
+
+		We want:
+			ah_check_after = sum_total_after(rc, bh, collectives, coretime, people)
+			ah_check_after = sum_total_after(bh, collectives, coretime, people) + rc_balance_kept
+		Use (3):
+			ah_check_after = sum_total_before(bh, collectives, coretime, people) + rc_balance_kept
+			ah_check_after = sum_total_before(ah, bh, collectives, coretime, people) - ah_total_before + rc_balance_kept
+		Use (1):
+			ah_check_after = rc_check_before - ah_total_before + rc_balance_kept
+		Use (0):
+			ah_check_after = `checking_balance` - ah_check_before - ah_total_before + rc_balance_kept
+			ah_check_after = `checking_balance` + rc_balance_kept - ah_total_before - ah_check_before
+		*/
 		// set it to the correct value:
-		let balance_after = rc_balance_before
-			.checked_sub(ah_issuance_before) // amount teleported to other system chains
+		let balance_after = checking_balance
+			.checked_add(rc_balance_kept)
+			.ok_or(Error::<T>::FailedToCalculateCheckingAccount)?
+			.checked_sub(balances_before.total_issuance)
 			.ok_or(Error::<T>::FailedToCalculateCheckingAccount)?
-			.checked_add(rc_balance_kept) // total amount teleported to other chains
+			.checked_sub(balances_before.checking_account)
 			.ok_or(Error::<T>::FailedToCalculateCheckingAccount)?;
 		<T as Config>::Currency::make_free_balance_be(&checking_account, balance_after);
+		<<T as Config>::Currency as Currency<_>>::deactivate(balance_after);
```

Comment:
we have two high ranks fellows voting for no deactivation for teleported assets - https://github.com/paritytech/polkadot-sdk/issues/8055

Reply:
yes, that's right, will remove